packages/@aws-cdk/cloud-assembly-schema/lib/assembly-manifest.ts

```diff
@@ -0,0 +1,340 @@
+     */
+    readonly repositoryName?: string;
+
+    /**
+     * The docker image tag to use for tagging pushed images. This field is
+     * required if `imageParameterName` is ommited (otherwise, the app won't be
+     * able to find the image).
+     *
+     * @default - this parameter is REQUIRED after 1.21.0
+     */
+    readonly imageTag?: string;
+
+    /**
+     * Build args to pass to the `docker build` command
+     *
+     * @default no build args are passed
+     */
+    readonly buildArgs?: { [key: string]: string };
+
+    /**
+     * Docker target to build to
+     *
+     * @default no build target
+     */
+    readonly target?: string;
+
+    /**
+     * Path to the Dockerfile (relative to the directory).
+     *
+     * @default - no file is passed
+     */
+    readonly file?: string;
+
+}
+
+/**
+ * @see ArtifactMetadataEntryType.ASSET
+ */
+export type AssetMetadataEntry = FileAssetMetadataEntry | ContainerImageAssetMetadataEntry;
+
+/**
+ * @see ArtifactMetadataEntryType.INFO
+ * @see ArtifactMetadataEntryType.WARN
+ * @see ArtifactMetadataEntryType.ERROR
+ */
+export type LogMessageMetadataEntry = string;
+
+/**
+ * @see ArtifactMetadataEntryType.LOGICAL_ID
+ */
+export type LogicalIdMetadataEntry = string;
+
+/**
+ * @see ArtifactMetadataEntryType.STACK_TAGS
+ */
+export type StackTagsMetadataEntry = Tag[];
+
+/**
+ * Union type for all metadata entries that might exist in the manifest.
+ */
+export type MetadataEntryData = AssetMetadataEntry | LogMessageMetadataEntry | LogicalIdMetadataEntry | StackTagsMetadataEntry;
+
+/**
+ * Type of artifact metadata entry.
+ */
+export enum ArtifactMetadataEntryType {
+
+    /**
+     * Asset in metadata.
+     */
+    ASSET = 'aws:cdk:asset',
+
+    /**
+     * Metadata key used to print INFO-level messages by the toolkit when an app is syntheized.
+     */
+    INFO = 'aws:cdk:info',
+
+    /**
+     * Metadata key used to print WARNING-level messages by the toolkit when an app is syntheized.
+     */
+    WARN = 'aws:cdk:warning',
+
+    /**
+     * Metadata key used to print ERROR-level messages by the toolkit when an app is syntheized.
+     */
+    ERROR = 'aws:cdk:error',
+
+    /**
+     * Represents the CloudFormation logical ID of a resource at a certain path.
+     */
+    LOGICAL_ID = 'aws:cdk:logicalId',
+
+    /**
+     * Tag metadata key.
+     */
+    STACK_TAGS = 'aws:cdk:stack-tags'
+}
+
+/**
+ * Type of cloud artifact.
+ */
+export enum ArtifactType {
+
+    /**
+     * Stub required because of JSII.
+     */
+    NONE = 'none', // required due to a jsii bug
+
+    /**
+     * The artifact is an AWS CloudFormation stack.
+     */
+    AWS_CLOUDFORMATION_STACK = 'aws:cloudformation:stack',
+
+    /**
+     * The artifact contains metadata generated out of the CDK application.
+     */
+    CDK_TREE = 'cdk:tree',
+}
+
+/**
+ * Key Value Tag.
+ */
+export interface Tag {
+
+    /**
+     * Tag key.
+     */
+    readonly key: string;
```

Comment:
2 spaces

Reply:
+1.

We should have a lint rule that enforces this (not as part of this PR). I'm having to comment this on community contributions as well.